stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -2173,14 +2176,22 @@ impl BitcoinRegtestController {
     /// * `true` if the transaction is confirmed (has at least one confirmation).
     /// * `false` if the transaction is unconfirmed or could not be found.
     pub fn is_transaction_confirmed(&self, txid: &Txid) -> bool {
-        match self.rpc_client.get_transaction(txid) {
+        match self
+            .rpc_client
+            .get_transaction(self.get_config_wallet(), txid)
+        {
             Ok(info) => info.confirmations > 0,
             Err(e) => {
                 error!("Bitcoin RPC failure: checking tx confirmation {e:?}");
                 false
             }
         }
     }
+
+    /// Returns the configured wallet name from [`Config`].
+    fn get_config_wallet(&self) -> &String {
```

Comment:
Nit:

```suggestion
    fn get_wallet_name(&self) -> &String {
```

I'm not totally sure about my suggestion, but IMO `get_config_wallet` sounds like something that will return a struct or implementation around the wallet, instead of just the name.

Reply:
I'm fine with your suggestion.

crc: c122976d4e18c5728837ab0b5de378bf2096613a